Content.Server/NPC/HTN/HTNSystem.cs

```diff
@@ -426,7 +424,7 @@ private void RequestPlan(HTNComponent component)
         if (component.PlanningJob != null)
             return;
 
-        component.PlanAccumulator += component.PlanCooldown;
+        component.NextPlanTime = _gameTiming.CurTime + TimeSpan.FromSeconds(component.PlanCooldown);
```

Comment:
This will eat the remainder with every update. Instead use
```suggestion
        component.NextPlanTime += TimeSpan.FromSeconds(component.PlanCooldown);
```
and `component.NextPlanTime = _timing.CurTime;` in a `MapInitEvent` subscription.

See #32805 for reference.

Reply:
What's wrong with eating the remainder? `NextPlanTime` is not something that is bumped consistently on a loop update cycle; it's only bumped when a new plan is selected. So the remainder is not meaningful.

Reply:
Isn't it bumped every `PlanCooldown` seconds?
Not sure how much precision matters here, but sloth said it should be generally always done that way if I understood it correctly. But maybe someone else can double check this, I'm not familiar with this system.

Reply:
`PlanCooldown` is used to prevent an NPC who has just gotten a new plan from instantly switching to a different one. While currently HTN will request a new plan immediately after the cooldown resulting in a bump every `PlanCooldown` seconds, that is not necessarily required and not necessarily the long term plan. But we can get input from @metalgearsloth 

I would agree with you about "precision" issues for a system that needs to update at a constant rate, but as far as I can tell this is really a cooldown and not a constant update system. Again we can ask @metalgearsloth to clarify the original intent of the code.